plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/newapp/EditValueDialog.java

```diff
@@ -0,0 +1,141 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors: Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.wizard.newapp;
+
+import org.eclipse.core.databinding.DataBindingContext;
+import org.eclipse.core.databinding.beans.BeanProperties;
+import org.eclipse.core.databinding.observable.value.IObservableValue;
+import org.eclipse.core.databinding.validation.MultiValidator;
+import org.eclipse.core.databinding.validation.ValidationStatus;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.jface.databinding.fieldassist.ControlDecorationSupport;
+import org.eclipse.jface.databinding.swt.WidgetProperties;
+import org.eclipse.jface.dialogs.IInputValidator;
+import org.eclipse.jface.dialogs.InputDialog;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.swt.widgets.Text;
+import org.jboss.tools.common.ui.databinding.ObservableUIPojo;
+import org.jboss.tools.common.ui.databinding.ValueBindingBuilder;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+
+@SuppressWarnings("rawtypes")
+public class EditValueDialog extends InputDialog {
+	static final String PROVIDE_REQUIRED_VALUE = "{0} is a required value, please provide a value.";
+	static final String PROVIDE_NEW_VALUE = "Please provide new value";
+
+	DataBindingContext dbc = new DataBindingContext();
+	InputModel model;
+
+	String name;
+	String initialValue;
+	boolean required;
+
+	public EditValueDialog(Shell shell, 
+			String title, String message, String name, String initialValue, boolean required) {
+		super(shell, title, message, initialValue, new InputValidator());
+		this.name = name;
+		this.initialValue = initialValue;
+		this.required = required;
+		model = new InputModel();
+	}
+
+	@Override
+	public InputValidator getValidator() {
+		return (InputValidator)super.getValidator();
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	protected Control createDialogArea(Composite parent) {
+		Composite control = (Composite)super.createDialogArea(parent);
+		Text text = null;
+		for (Control c: control.getChildren()) {
```

Comment:
this code would be a nice candidate for a method #getTextInput(Control[] controls)